markup/forms.md

```diff
@@ -0,0 +1,71 @@
+# Forms
+
+## Checklist
+
+### Buttons rather than inputs
+
+- Specify the required attributes (name and value)
+
+### Multistep forms over multiple submit buttons
+
+### Server-side Validation
+
+- Start with server-side validation
+- Return an error message for each erroring form element.
+
+### Client-side validation
+
+### Accessibility
+
+### Label every input
+
+### Use the most simple form control that does the job
+
+### Minimise the number of fields to fill out
+
+### Mark optional fields
+
+### Size according to the expected input
+
+### Retain focus for every input element
+
+- For preference, enhance it! Make it part of the brand!
+
+### Don't ask users to repeat their email address
+
+### Provide a "show password" option
+
+### Don't split data fields
+
+### Avoid dropdown menus
+
+### Provide examples of input
+
+- Don't use placeholders!
+
+### Offer help text
+
+### Masked input
```

Comment:
I thought this to be a good practice until I read something that challenged it. The reasons do not come out from the top of my head atm, therefore I would research that one. Though Estelle Weyl has a great [framework agnostic accessible input masking library](https://github.com/estelle/input-masking/)